src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/ConstructorSymbolReferenceFinder.cs

```diff
@@ -146,5 +147,55 @@ protected override bool CanFind(IMethodSymbol symbol)
                 ? FindReferencesInDocumentUsingIdentifierAsync(symbol, simpleName, document, semanticModel, cancellationToken)
                 : new ValueTask<ImmutableArray<FinderLocation>>(ImmutableArray<FinderLocation>.Empty);
         }
+
+        private Task<ImmutableArray<FinderLocation>> FindReferencesInImplicitObjectCreationExpressionAsync(
+            IMethodSymbol symbol,
+            Document document,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            // Only check `new (...)` calls that supply enough arguments to match all the required parameters for the constructor.
+            var minimumArgumentCount = symbol.Parameters.Count(p => !p.IsOptional && !p.IsParams);
+            var maximumArgumentCount = symbol.Parameters.Length > 0 && symbol.Parameters.Last().IsParams
```

Comment:
Consider avoiding LINQ:

```suggestion
            var maximumArgumentCount = symbol.Parameters.Length > 0 && symbol.Parameters[^1].IsParams
```

Reply:
this should not be linq: https://docs.microsoft.com/en-us/dotnet/api/system.linq.immutablearrayextensions.last?view=netcore-3.1